src/main/java/com/dataloom/requests/HazelcastRequestsManager.java

```diff
@@ -82,4 +111,52 @@ public void submitAll( Map<AceKey, Status> statusMap ) {
     public Stream<Status> getStatuses( Stream<AceKey> requestKeys ) {
         return requestKeys.map( Util.getSafeMapper( requests ) );
     }
+
+    private void signalPermissionRequestStatusUpdates( Map<AceKey, Status> aceKeyRequestStatusMap ) {
```

Comment:
Also, I think you could just write EntryListeners and accomplish something similar without the added complexity of having signal code side by side.